### ISSUE TITLE
API is not returning all SCA checks for a given policy

### ISSUE THREAD

**Issue by mgmacias95** [wazuh/wazuh#3062]
Hello team,

These are my checked policies:
```javascript
# curl -u foo:bar "localhost:55000/sca/000?pretty&select=pass,fail,policy_id"
{
   "error": 0,
   "data": {
      "totalItems": 3,
      "items": [
         {
            "pass": 3,
            "fail": 6,
            "policy_id": "system_audit_ssh"
         },
         {
            "pass": 76,
            "fail": 0,
            "policy_id": "system_audit"
         },
         {
            "pass": 36,
            "fail": 5,
            "policy_id": "cis_debian"
         }
      ]
   }
}
```

As you can see, I have 76 checks from `system_audit` policy but when I request those checks, only 16 are returned:
```javascript
# curl -u foo:bar "localhost:55000/sca/000/checks/system_audit?pretty"
{
   "error": 0,
   "data": {
      "totalItems": 16,
      "items": [
         ...
            ]
         }
      ]
   }
}
```

Best regards,
Marta

**Comment by mgmacias95** [comment]
At first I thought the API was returning the wrong information since we join multiple SQL tables and that could lead to wrong data but I have just checked the SCA database and there's the same issue there:
```sql
sqlite> select pass,fail,policy_id from sca_scan_info;
36|5|cis_debian
76|0|system_audit
3|6|system_audit_ssh
sqlite> select count(*) from sca_check where policy_id='system_audit';
16
sqlite> select count(*) from sca_check where policy_id='system_audit_ssh';
9
sqlite> select count(*) from sca_check where policy_id='cis_debian';
41
sqlite> select id, policy_id from sca_scan_info;
406275960|cis_debian
723522447|system_audit
1346881125|system_audit_ssh
```

Can any member of @wazuh/core light me here? Thanks!

**Comment by chemamartinez** [comment]
Merged: #3074
